cpp/src/rolling/grouped_rolling.cu

```diff
@@ -715,115 +720,161 @@ std::unique_ptr<column> time_range_window_DESC(
-                                      mr);
+    return group_offsets.is_empty() ? range_window_DESC(input,
+                                                        orderby_column,
+                                                        preceding_value,
+                                                        preceding_window.is_unbounded(),
+                                                        following_value,
+                                                        following_window.is_unbounded(),
+                                                        min_periods,
+                                                        aggr,
+                                                        stream,
+                                                        mr)
+                                    : range_window_DESC(input,
+                                                        orderby_column,
+                                                        group_offsets,
+                                                        group_labels,
+                                                        preceding_value,
+                                                        preceding_window.is_unbounded(),
+                                                        following_value,
+                                                        following_window.is_unbounded(),
+                                                        min_periods,
+                                                        aggr,
+                                                        stream,
+                                                        mr);
   }
 }
 
+struct dispatch_grouped_range_rolling_window {
+  template <typename OrderByColumnType, typename... Args>
+  std::enable_if_t<!detail::is_supported_order_by_column_type<OrderByColumnType>(),
+                   std::unique_ptr<column>>
+  operator()(Args&&...) const
+  {
+    CUDF_FAIL("Unsupported OrderBy column type.");
+  }
+
+  template <typename OrderByColumnType, typename... Args>
+  std::enable_if_t<detail::is_supported_order_by_column_type<OrderByColumnType>(),
+                   std::unique_ptr<column>>
+  operator()(Args&&... args) const
+  {
+    return grouped_range_rolling_window_impl<OrderByColumnType>(std::forward<Args>(args)...);
+  }
+};
+
+struct to_duration_bounds {
+  template <typename OrderBy, std::enable_if_t<cudf::is_timestamp<OrderBy>(), void>* = nullptr>
+  range_window_bounds operator()(size_type num_days) const
+  {
+    using DurationT = typename OrderBy::duration;
+    return range_window_bounds::get(duration_scalar<DurationT>{duration_D{num_days}, true});
+  }
+
+  template <typename OrderBy, std::enable_if_t<!cudf::is_timestamp<OrderBy>(), void>* = nullptr>
+  range_window_bounds operator()(size_type num_days) const
+  {
+    CUDF_FAIL("Expected timestamp orderby column.");
+  }
+};
+
+range_window_bounds to_range_bounds(cudf::size_type num_days, cudf::data_type timestamp_type)
```

Comment:
Docs

Reply:
Point taken.
(I was trying to be sneaky about not documenting short utility/implementation functions in the `unnamed` namespace.)